### PR TITLE
fix broken link -- js-polyfill is a 404

### PR DESCRIPTION
**Pull request by NukeManDan**
I can't find this in the source presently to replace it: the link to `js-polyfill` is a 404
Can someone put a working link?

### CODE REVIEW

**Comment by sunfishcode** [comment]
The original JS polyfill was built from compiling the initial WASI implementation in Wasmtime written in C with Emscripten. Since then, Wasmtime has moved to an implementation written in Rust and dropped the C version. In theory, it should be possible to compile the Rust code with Emscripten in the same way, however at this time, this has not yet been done.

One lesson learned is that compiling with Emscripten tends to produce a lot of code, while hand-written approaches, such as [this one](https://github.com/bjorn3/rust/blob/compile_rustc_wasm4/rustc.html) or [this](https://github.com/WebAssembly/waterfall/blob/master/src/wasi.js) tend to be a lot smaller and more flexible. Instead of Implementing WASI on top of a Posix-ish API and bundling an implementation of the Posix-ish API to support that, they can just directly implement the WASI APIs.

At this point, we should probably remove the mentions of the original JS polyfill. Its main purpose was to demonstrate that it's possible to polyfill WASI on the Web, which it did, and now there are better alternatives. The two I linked to above aren't complete, but they're enough for a fair number of use cases.

**Comment by NukeManDan** [comment]
@sunfishcode I wanna close this one out, is there a more recent PR that does what you describe to remove and rework this section? Thanks! 😁

**Comment by sunfishcode** [comment]
No, there isn't yet a new PR for this unfortunately. A lot of parts of WASI have changed, and are still in the process of actively changing. If you want to make the minimal fix here, the most recent version of the original js-polyfill code is https://github.com/bytecodealliance/wasmtime/tree/177af5357880c33863cbe3363754000dbd23bddb/crates/wasi-c.

File: docs/WASI-intro.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ or `cargo run --bin wasmtime foo.wasm`.
 
 The polyfill is online [here](https://wasi.dev/polyfill/).
 
-The source is [here](https://github.com/bytecodealliance/wasmtime/tree/main/crates/wasi-c/js-polyfill).
+The source is [here](#FIXME).
```
Comment (NukeManDan) [inline docs/WASI-intro.md:57]:
```suggestion
The source is [here](https://github.com/bytecodealliance/wasmtime/tree/177af5357880c33863cbe3363754000dbd23bddb/crates/wasi-c).
```

**Pull request closed without merging**